Add graphite basicauth doc (#445)
kedacore/keda-docs: content/docs/2.3/scalers/graphite.md

+++
title = "Graphite"
layout = "scaler"

> availability = "v1.0+"

Reviewer: ```suggestion
availability = "v2.3+"
```

maintainer = "Community"
description = "Scale applications based on Graphite."
go_file = "graphite_scaler"
+++

### Trigger Specification

This specification describes the `graphite` trigger that scales based on a Graphite.

```yaml
triggers:
- type: graphite
  metadata:
    # Required
    serverAddress: http://<graphite-host>:81
    metricName: request-count
    query: stats.counters.http.hello-world.request.count.count # Note: query must return a vector/scalar single element response
    threshold: '100'
    queryTime: '-10Minutes'
```

**Parameter list:**

- `serverAddress` - Address of Graphite
- `metricName` - Metric name to use

> - `query` - Query to run

Reviewer: Can you expand this a bit please? I see the following above for example: "# Note: query must return a vector/scalar single element response"

Reviewer: Would be nice to have more explanation for the other fields as well

- `threshold` - Value to start scaling for
- `queryTime` - Query Time to from 

> 

Reviewer: We typically have a "Authentication Parameters" section describing what is supported/required, please have a look at our other scalers.

### Example

```yaml
apiVersion: keda.sh/v1alpha1
kind: ScaledObject
metadata:
  name: graphite-scaledobject
  namespace: default
spec:
  scaleTargetRef:
    name: my-deployment
  triggers:
  - type: graphite
    metadata:
      serverAddress: http://<graphite-host>:9090
      metricName: request-count
      threshold: '100'
      query: stats.counters.http.hello-world.request.count.count
      queryTime: '-10Minutes'
```

> 
> 

Reviewer: ```suggestion

```

> Here is an example of a graphite scaler with Basic Authentication, define the `Secret` and `TriggerAuthentication` as follows

Reviewer: ```suggestion
Here is an example of scaling with the Graphite scaler with basic authentication by defining the `Secret` and `TriggerAuthentication` as follows:
```
